### ISSUE TITLE
Lets owerrite array's attributes for list

### ISSUE THREAD

**Issue by levBagryansky** [objectionary/eo-collections#28]
@yegor256 , @Graur 
Imagine situation: I have two lists and I need to cancat them and get this list with added one more object. So to get list after that I must wite
```
[list1 list2 item] > l-concat-with
  list > @
    without.
      list
        concat.
          list1
          list2
      item
```
instead of simple
```
[list1 list2 i] > l-concat-with
  without. > @
    concat.
      list1
      list2
    i
```
I think its wrong that after `array`'s attribute list returns `array`. It have to return a `list`.
What do you think?

**Comment by Graur** [comment]
@levBagryansky I think you can write like this, since `list` decorates `array` and already has all of its objects.
Moreover, this test passes:
```
[] > concat-of-two-lists
  list (* 1 2) > list1
  list (* 1 2) > list2
  with. > list3
    concat.
      list1
      list2
    3
  assert-that > @
    list3.at 4
    $.equal-to 3
```

**Comment by levBagryansky** [comment]
@Graur , 
my guilt, I ment you cannot use list's attributes after that. I must be without instead of with.

**Comment by Graur** [comment]
@levBagryansky Indeed, we need to return `list` in such cases. We have already talk about it here: https://github.com/objectionary/eo/issues/807#issuecomment-1174116639. Please, do it

**Comment by Graur** [comment]
@yegor256 Can you please tell us, should `lis.with` be the object either `array` or `list`?

**Comment by yegor256** [comment]
@Graur I think it should be `list`

**Comment by levBagryansky** [comment]
Ok, I implemented it in #29
